src/components/app-pagination/AppPagination.tsx

```diff
@@ -1,17 +1,29 @@
+import { ChangeEvent, FC } from 'react'
+
 import Pagination from '@mui/material/Pagination'
 import Box from '@mui/system/Box'
+
+import { Size } from '~/types'
 import { styles } from '~/components/app-pagination/AppPagination.styles'
 
-const AppPagination = ({
-  size = 'medium',
+interface AppPaginationProps {
+  size?: Size
+  page: number
+  itemsCount: number
+  pageSize: number
+  setCurrentPage: (page: number) => void
+}
+
+const AppPagination: FC<AppPaginationProps> = ({
+  size = Size.Medium,
   page,
   itemsCount,
   pageSize,
   setCurrentPage
```

Comment:
onChange instead of setCurrentPage. also add posibility to pass additional props to pagination, we don`t need to specify page in props it will go in additional props

Reply:
Added additional props and changed onChange function naming